shared/middleware/net_sim_api.rb

```diff
@@ -69,6 +76,40 @@ def has_json_utf8_headers(request)
     get_table(shard_id, table_name).fetch(id.to_i).to_json
   end
 
+  # Delete all wires associated with (locally or remotely) a given node_id
+  #
+  # @param [String] shard_id
+  # @param [Integer] node_id
+  def delete_wires_for_node(shard_id, node_id)
+    wire_table = get_table(shard_id, TABLE_NAMES[:wire])
+    wire_ids = wire_table.to_a.select do |wire|
+      wire['localNodeID'] == node_id or wire['remoteNodeID'] == node_id
+    end.map do |wire|
+      wire['id']
+    end
+    wire_ids.each do |wire_id|
+      wire_table.delete(wire_id)
+    end
+    wire_ids
+  end
+
+  # Delete all messages simulated by a given node_id
+  #
+  # @param [String] shard_id
+  # @param [Integer] node_id
+  def delete_messages_for_node(shard_id, node_id)
+    message_table = get_table(shard_id, TABLE_NAMES[:message])
+    message_ids = message_table.to_a.select do |message|
+      message['simulatedBy'] == node_id
+    end.map do |message|
```

Comment:
"end.map" looks funny. This is easier to read if you use curly braces.  Also, you can write the map more concisely as follows:

message_table.to_a.select {|msg| msg['simulatedBy']}.map(:id)